sdk/core/src/policies/telemetry_policy.rs

```diff
@@ -62,3 +85,62 @@ impl Policy for TelemetryPolicy {
         next[0].send(ctx, request, &next[1..]).await
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    // tests assume cargo + rustc
+    const CRATE_NAME: &'static str = env!("CARGO_PKG_NAME");
+    const CRATE_VERSION: &'static str = env!("CARGO_PKG_VERSION");
+    const RUSTC_VERSION: &'static str = env!("AZSDK_RUSTC_VERSION");
+
+    struct EmptyEnv;
+    impl Environment for EmptyEnv {
+        fn crate_name() -> Option<&'static str> {
+            None
+        }
+
+        fn crate_version() -> Option<&'static str> {
+            None
+        }
+
+        fn rustc_version() -> Option<&'static str> {
+            None
+        }
+    }
+
+    #[test]
+    fn test_default() {
```

Comment:
These tests seem really close to simply reimplementing the functionality it's testing which IMO is worse than having no tests at all. I'm fine with keeping these, but I'm really not convinced of their utility.

Reply:
To make sure we get repeated and consistent output. We have unit tests for these in our other repos as well.